.github/workflows/build.yml

```diff
@@ -28,18 +27,19 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - run: python3 script/check_release.py --version ${{ env.version }} --target ${{ matrix.target }} --build-type ${{ matrix.build_type }} --machine ${{ matrix.machine }}
-        if: ${{ github.event.inputs.skip_release != 'true' }}
+        if: ${{ github.event.inputs.skip_release != 'true' && github.ref == 'refs/heads/main' }}
```

Comment:
If I understand correctly, above it's configured to run for pull-requests into the main branch:
```
pull_request:
    branches:
      - main
```

But here, `&& github.ref == 'refs/heads/main'` restricts the jobs to run only in the main branch. Right?
Sometimes it's convenient to run the checks (build and tests) without merging into master, so in a separate branch or in PR.  Can we support it? What do you think?

Reply:
yes, we only skip publishing artifacts, if we build from non-main branch.